crates/matrix-sdk-crypto/src/store/mod.rs

```diff
@@ -920,6 +922,15 @@ pub trait CryptoStore: AsyncTraitDeps {
     /// * `request_id` - The unique request id that identifies this outgoing key
     /// request.
     async fn delete_outgoing_secret_requests(&self, request_id: &TransactionId) -> Result<()>;
+
+    /// TODO: docs
+    async fn load_encryption_settings(
```

Comment:
One thing I mentioned already, but it is still my main concern with this patch is that this is expanding the `CryptoStore` trait. I prefer to avoid this if possible, especially since this functionality is not something the `matrix-sdk` crate will use.

The idea here is that we might want to reuse the existing `get`/`set` value methods the crypto stores have. This would also allow people to store other custom things.

Reply:
Yea that makes sense, I'll rework it as a separate commit so we can compare it

Reply:
> The idea here is that we might want to reuse the existing `get`/`set` value methods the crypto stores have. This would also allow people to store other custom things.

To be clear, we would still need to expose some something like `async fn set_value(&self, key: &str, value: &impl Serialize) -> Result<()>;` (and getter equivalent) in the `CryptoStore` trait, at which point the compiler becomes unhappy about the use of generics when trying to make an object from the trait. Is this what you are suggesting?

Reply:
Yep, the state store already has such an interface, though it puts the responsibility of serializing onto the caller:
https://github.com/matrix-org/matrix-rust-sdk/blob/7e64c15145ec7a5d7c21478e4a0f38ecbff29fca/crates/matrix-sdk-sled/src/state_store.rs#L1245

Of course the interface of the `CryptoStore` needs to be exposed on the `OlmMachine` where you can put the generic `impl Serialize` argument if you prefer the Rust side to take care of the serialization.